packages/core-utils/src/format/number-format.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Format a size in bytes for display.
+ *
+ * Formats the size in kibibytes, mebibtyes, gibibytes, etc.
+ */
+export const formatBytes = (bytes: number): string => {
+  const prefixes: [string, number][] = [
+    ['P', 2 ** 50],
+    ['T', 2 ** 40],
+    ['G', 2 ** 30],
+    ['M', 2 ** 20],
+    ['K', 2 ** 10],
+  ];
+  const maybePrefixAndDivisor = prefixes.find(([_prefix, divisor]) => bytes >= divisor);
+  if (maybePrefixAndDivisor) {
+    const [prefix, divisor] = maybePrefixAndDivisor;
+    return `${(bytes / divisor).toPrecision(3)} ${prefix}iB`;
+  }
+  return `${bytes} B`;
+};
+
+const numberFormatter = new Intl.NumberFormat('en-US');
```

Comment:
Hard-coding the locale kind of defeats the purpose of using `Intl`, but I didn't want to change behavior in this PR.